resourcemanager/commonids/composite_resource_id_test.go

```diff
@@ -5,9 +5,41 @@ package commonids
 
 import (
 	"fmt"
+	"strings"
 	"testing"
 )
 
+func TestNewCompositeResourceID(t *testing.T) {
+	botIdString := "/subscriptions/12345678-1234-9876-4563-123456789012/resourceGroups/example-resource-group/providers/Microsoft.BotService/botServices/botServiceValue"
+	appIdString := "/subscriptions/12345678-1234-9876-4563-123456789012/resourceGroups/example-resource-group/providers/Microsoft.Web/sites/siteValue"
+
+	botId, err := ParseBotServiceID(botIdString)
+	if err != nil {
+		t.Fatalf("parsing resource ID for composite resource ID test: %+v", err)
+	}
+
+	appId, err := ParseAppServiceID(appIdString)
+	if err != nil {
+		t.Fatalf("parsing resource ID for composite resource ID test: %+v", err)
+	}
+
+	id := NewCompositeResourceID(botId, appId)
+
+	if !strings.EqualFold(id.First.ID(), botIdString) {
+		t.Fatalf("expected First ID string to be %q but got %q", botIdString, id.First.ID())
+	}
+
+	if !strings.EqualFold(id.Second.ID(), appIdString) {
+		t.Fatalf("expected Second ID string to be %q but got %q", appIdString, id.Second.ID())
+	}
+
+	expectedIdString := "/subscriptions/12345678-1234-9876-4563-123456789012/resourceGroups/example-resource-group/providers/Microsoft.BotService/botServices/botServiceValue|/subscriptions/12345678-1234-9876-4563-123456789012/resourceGroups/example-resource-group/providers/Microsoft.Web/sites/siteValue"
+
+	if !strings.EqualFold(id.ID(), expectedIdString) {
```

Comment:
this should exact match?

```suggestion
	if id.ID != expectedIdString {
```

Reply:
you're right, sorry habit